### DISCUSSION TITLE
Repo Errors NO_PUBKEY

### DISCUSSION THREAD

**Discussion by ScanCaster** [Xpra-org/xpra#3960] [Q&A] [answered]
With all the recent changes on the repos etc... I am getting this when attempting to update the existing installs

```
Get:5 https://xpra.org focal InRelease [4,734 B]                                                                                                                           
Err:5 https://xpra.org focal InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 73254CAD17978FAF
Fetched 119 kB in 26s (4,579 B/s)
Reading package lists... Done

Get:14 https://xpra.org jammy InRelease [4,734 B]                                                                                                                          
Err:14 https://xpra.org jammy InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 73254CAD17978FAF
Fetched 2,872 kB in 26s (112 kB/s)
Reading package lists... Done

W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: https://xpra.org jammy InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 73254CAD17978FAF
W: Failed to fetch https://xpra.org/dists/jammy/InRelease  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 73254CAD17978FAF
W: Some index files failed to download. They have been ignored, or old ones used instead.


```
Applies to 20.04, 22.04, 23.04 via DragonOS

I've tried adding the "new" keys via :
https://github.com/Xpra-org/xpra/wiki/Download#-linux

Would like to update the installs with the new versions. Thanks.

**Comment by ScanCaster** [comment]
Well at least for 20.04 I found and used this, to solve it

edit /etc/apt/sources.list

and use this line for the XPRA Repo

`deb [arch=amd64,arm64 signed-by=/usr/share/keyrings/xpra.asc] https://xpra.org/ focal main`

No joy on this for 23.04 as it appears to use some other format and this line is not in sources.list

**Comment by ScanCaster** [comment]
No joy on this with 23.04 so will have to investigate further on that....

**Comment by totaam** [comment, marked as answer]
My guess is that you already had a file named `/usr/share/keyrings/xpra.asc` on that system and for some unknown reason, the Debian tools will not re-import this file when it is modified.
Try renaming it... ie: `xpra-new.asc`

If there is a way to tell `apt` to reload the updated key, I haven't found it.

**Comment by goekce** [comment]
I also could not find a way to reload the key :/ Currently you have to change the name of the key as @totaam stated.

If you are anyway on it you can also put the key into the new recommended location on Ubuntu. Here is an excerpt from `man apt-key` in Ubuntu 22.04:

> Recommended: Instead of placing keys into the /etc/apt/trusted.gpg.d directory, you can place them anywhere on your filesystem by using the Signed-By option in your sources.list and pointing to the filename of the key. See sources.list(5) for details. Since APT 2.4, /etc/apt/keyrings is provided as the recommended location for keys not managed by packages.

So instead of `/use/share/keyrings/...` use `/etc/apt/trusted.gpg.d` and change the key path in `/etc/apt/sources.list.d/xpra.sources` accordingly. Then you can delete the old path `/usr/share/keyrings/xpra.asc`.
